data/transactions/verify/txn.go

```diff
@@ -100,6 +100,9 @@ func TxnPool(s *transactions.SignedTxn, ctx Context, verificationPool execpool.B
 	if s.Txn.Src() == zeroAddress {
 		return errors.New("empty address")
 	}
+	if !proto.SupportRekeying && (s.AuthAddr != basics.Address{}) {
```

Comment:
Since `WellFormed` now does this check, and we call it a few lines above, can we now drop this check from here?

Reply:
(I think we can take out the `if s.Txn.Src` check above, for that matter.)

Reply:
WellFormed checks the `RekeyTo` but it doesn't (yet) check the `AuthAddr` -- in this PR `WellFormed` still takes in a `Transaction` rather than a `SignedTxn` so it can't see the `AuthAddr` field. (I'm going to change that in a separate PR as Max suggested earlier, as it'll involve changing a bunch of the transaction tests.)

Reply:
Ah, right, I missed that. Looks good then.